apollo-gradle-plugin-incubating/src/main/kotlin/com/apollographql/apollo/gradle/ServiceVariant.kt

```diff
@@ -0,0 +1,143 @@
+
+      return ServiceVariant(
+          name = service.name,
+          files = files,
+          schemaFile = schemaFile,
+          schemaPackageName = schemaPackageName,
+          rootPackageName = service.rootPackageName ?: ""
+      )
+    }
+
+    fun default(project: Project, sourceSetNames: List<String>): List<ServiceVariant> {
+      val schemaFiles = findFilesInSourceSets(project, sourceSetNames, ".") {
+        it.name == "schema.json"
+      }
+
+      if (schemaFiles.isEmpty()) {
+        return emptyList()
+      }
+
+      var i = 0
+      val services = schemaFiles.entries
+          .sortedBy { it.value.canonicalPath } // make sure the order is predicable for tests
+          .map { entry ->
+        val sourceFolderPath = entry.value.canonicalPath.relativePathToGraphql(dropLast = 1)!!
+        val files = findFilesInSourceSets(project, sourceSetNames, sourceFolderPath, ::isGraphQL).values.toList()
+
+        val name = (i++).toString()//entry.key.split(File.separator).map { it.capitalize() }.joinToString("")
+
+        ServiceVariant(
+            name = name,
+            files = files,
+            schemaFile = entry.value,
+            schemaPackageName = entry.value.canonicalPath.formatPackageName(dropLast = 1)!!,
+            rootPackageName = ""
+        )
+      }
+
+      return services
+    }
+
+    fun isGraphQL(file: File): Boolean {
+      return file.name.endsWith(".graphql") || file.name.endsWith(".gql")
+    }
+
+    /**
+     * Finds the files in the given sourceSets taking into account their precedence according to the android plugin order
+     *
+     * Returns a map with the relative path to the path as key and the file as value
+     *
+     * Files coming last will have higher priorities that the first ones.
+     */
+    private fun findFilesInSourceSets(project: Project, sourceSetNames: List<String>, path: String, predicate: (File) -> Boolean): Map<String, File> {
+      val candidates = mutableMapOf<String, File>()
+      sourceSetNames.forEach { sourceSetName ->
+        val root = project.projectDir.child("src", sourceSetName, "graphql", path)
+        val files = findFiles(root, predicate)
+
+        files.forEach {
+          val key = if (root.isFile) {
+            // toRelativeString only works on directories.
+            ""
+          } else {
+            it.toRelativeString(root)
+          }
+
+          // overwrite the previous entry if it was there already
+          // this is ok as Android orders the sourceSetNames from lower to higher priority
+          candidates[key] = it
+        }
+      }
+      return candidates
+    }
+
+    private fun findFiles(root: File, predicate: (File) -> Boolean): List<File> {
+      val files = mutableListOf<File>()
+      if (root.exists()) {
+        findFiles(root, files, predicate)
+      }
+      return files
+    }
+
+    private fun findFiles(root: File, output: MutableList<File>, predicate: (File) -> Boolean) {
```

Comment:
nit: can this function just return a list of files instead of taking output as argument?
```
private inline fun File.findFiles(predicate: (File) -> Boolean): List<File> {
  when {
    isDirectory -> listFiles()?.flatMap { it.findFiles(predicate) } ?: emptyList()
    isFile && predicate(this) -> listOf(this)
    else -> emptyList()
  }
}
```